src/api/procedures/__tests__/modifyAllowance.ts

```diff
@@ -0,0 +1,206 @@
+import { AccountId, Balance } from '@polkadot/types/interfaces';
+import BigNumber from 'bignumber.js';
+import sinon from 'sinon';
+
+import { getAuthorization, prepareModifyAllowance } from '~/api/procedures/modifyAllowance';
+import { Context, ModifyAllowanceParams, Subsidy } from '~/internal';
+import { TxTags } from '~/polkadot';
+import { dsMockUtils, entityMockUtils, procedureMockUtils } from '~/testUtils/mocks';
+import { Mocked } from '~/testUtils/types';
+import { AllowanceOperation } from '~/types';
+import * as utilsConversionModule from '~/utils/conversion';
+
+jest.mock(
+  '~/api/entities/Account',
+  require('~/testUtils/mocks/entities').mockAccountModule('~/api/entities/Account')
+);
+
+jest.mock(
+  '~/api/entities/Subsidy',
+  require('~/testUtils/mocks/entities').mockSubsidyModule('~/api/entities/Subsidy')
+);
+
+describe('modifyAllowance procedure', () => {
+  let mockContext: Mocked<Context>;
+  let subsidy: Subsidy;
+  let stringToAccountIdStub: sinon.SinonStub<[string, Context], AccountId>;
+  let bigNumberToBalanceStub: sinon.SinonStub<
+    [BigNumber, Context, (boolean | undefined)?],
+    Balance
+  >;
+  let args: ModifyAllowanceParams;
+  let allowance: BigNumber;
+  let rawBeneficiaryAccountId: AccountId;
+  let rawAllowance: Balance;
+
+  let increasePolyxLimitTransaction: sinon.SinonStub;
+
+  beforeAll(() => {
+    dsMockUtils.initMocks();
+    procedureMockUtils.initMocks();
+    entityMockUtils.initMocks();
```

Comment:
Probably better to explicitly set the current allowance, it's a bit more readable